libs/agno/agno/agent/agent.py

```diff
@@ -1553,6 +1559,8 @@ def get_tools(self) -> Optional[List[Union[Toolkit, Callable, Function, Dict]]]:
         if self.knowledge is not None or self.retriever is not None:
             if self.search_knowledge:
                 agent_tools.append(self.search_knowledge_base)
+            if self.asearch_knowledge:
```

Comment:
Rather pass a `async_mode:bool` to this function to know whether it is being called from `arun` or `run`